client.js

```diff
@@ -122,7 +123,8 @@ class Client extends EventEmitter {
 		const id = ~~(Math.random() * 900) + 100;
 		const chars = 'abcdefghijklmnopqrstuvwxyz0123456789_'.split('');
 		const str = Array.from({ length: 8 }, () => chars[~~(Math.random() * 36)]).join('');
-		const conStr = `wss://${this.opts.server}${this.opts.port ? `:${this.opts.port}` : ''}/showdown/${id}/${str}/websocket`;
+		const conStr = `${this.opts.serverProtocol}://${this.opts.server}\
+${this.opts.port ? `:${this.opts.port}` : ''}/showdown/${id}/${str}/websocket`;
```

Comment:
Could this all be destructured out on the line before (eg: `const { protocol, port = ':443' } = this.opts`) so it doesn't overflow into multi-line template literal? Having the next line be non-indented is less cute

Reply:
Did